packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -88,7 +100,14 @@ async def unarchive_dir(archive_to_extract: Path, destination_folder: Path) -> N
                 for zip_entry in zip_file_handler.infolist()
             ]
 
-            await asyncio.gather(*tasks)
+            extracted_paths: List[Path] = await asyncio.gather(*tasks)
+
+            # NOTE: extracted_paths includes all tree leafs, which might include files and empty folders
+            return set(
+                p
+                for p in extracted_paths
+                if p.is_file() or (p.is_dir() and not any(p.glob("*")))
```

Comment:
please rename p to something more readable at this point.

Reply:
I find this very readable :-)   if "is a file or a directory with nothing" maps nicely, oder? :-)